server/index.js

```diff
@@ -43,39 +55,54 @@ app.get('/get_package_info', (req, res) =>{
 
 
 /* ***** add_package  *****
-    desc: add a package to start tracking
+    desc: add a package to start tracking. 3 step mechanism
+      1. Validate user 
+      2. Add package to start tracking in PKGE
+      3. Add package reference to user in Firestore
     inputs: 
       trackingNumber: tracking number of the package to track
     outputs: 
       code 200: successfully adds the new package
       code 400: unable to add package, more error codes and error messages in the return response
 */ 
 app.post('/add_package', (req, res) => {
-  if(req.query.trackingNumber == undefined){
+  const trackingNumber = req.query.trackingNumber;
+  if(trackingNumber == undefined){
     res.status(400).send("Requires a tracking number!");
   }
+  if(req.query.userId == undefined){
+    res.status(400).send("Requires a user ID!");
+  }
 
-  axios({
-    method: 'post', 
-    url: 'https://api.pkge.net/v1/packages?',
-    params: {
-      trackNumber: req.query.trackingNumber,
-      courierId: -1
-    },
-    headers: {
-      'Accept': 'application/json',
-      'X-Api-Key': process.env.PKGE_API_KEY
-    }
-  }).then((response) => {
-    res.status(200).send(JSON.stringify(response.data))
+  //check to see if user exists
+  auth.getAuth().getUser(req.query.userId).then((user) => {
+    //user exists, proceed to add package to PKGE 
+    pkgeAPI.addPackage(trackingNumber).then((r) => {
+      // package added to PKGE, proceed to make reference in firestore db
+      const docRef = db.collection(COLLECTION_NAME).doc(req.query.userId).collection('trackings').doc(trackingNumber);
+      docRef.set({
+        'status': 'started tracking'
```

Comment:
What are the different statuses? Also could we use camelCase format or rename it into one word?

Reply:
I've added all the delivery statuses